### PR TITLE
don't create DB until checks on stopcp are completed

### LINKED ISSUE #4637: Giving an invalid `--stopcp` on start "corrupts" database
If you have a not yet run workflow and give an invalid value for `--stopcp` when starting it for the first time, after the workflow shuts down due to the error, you are left with a workflow database with empty tables.

Because the `workflow_params` table is empty, if you try to restart the workflow, Cylc thinks the database is corrupted and refuses to play ball.

```
ERROR - Workflow shutting down - ServiceFileError: Cannot restart - Workflow database is incompatible with Cylc 8.0rc1.dev, or is corrupted
```
And you can't clean the database because of #4450 
```
$ cylc clean myflow/ --rm .service
ServiceFileError: Cannot clean - Workflow database is incompatible with Cylc 8.0rc1.dev, or is corrupted
```

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by wxtim**
These changes close #4637 

Don't create a workflow database until after validating stopcp.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [ ] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by MetRonnie** [comment]
The answer to #4709 may affect this

File: cylc/flow/scheduler.py [hunk 1]
```diff
@@ -121,6 +122,9 @@
 from cylc.flow.xtrigger_mgr import XtriggerManager
 
 
+MODE_INT = 'integer'
```
Comment (oliver-sanders) [inline cylc/flow/scheduler.py:125]: Can get this from:

```python
cylc.flow.cycling.loader.INTEGER_CYCLING_TYPE
```

Comment (wxtim) [reply cylc/flow/scheduler.py:125]: I thought I'd grepped for it.

**Comment by wxtim** [comment]
@oliver-sanders  - somehow I'd forgotten about this PR ...

**Comment by MetRonnie** [comment]
> The answer to #4709 may affect this

If #4709 is a bug the way I think it is, then this would be even easier. The stopcp processing logic could get moved out of scheduler.py and into config.py where its brethren already exist

**Comment by wxtim** [comment]
Closed by https://github.com/cylc/cylc-flow/issues/4709#issuecomment-1072340816

**Pull request closed without merging**